lib/nostrum/struct/message/component.ex

```diff
@@ -152,6 +161,35 @@ defmodule Nostrum.Struct.Message.Component do
   """
   @type max_values :: 1..25 | nil
 
+  @typedoc """
+  Minimum length of the input text.
+
+  Defaults to ``0``. Maximum of ``4000``. Only present for text inputs.
+  """
+  @type min_length :: 0..4000 | nil
+
+  @typedoc """
+  Maximum length of the input text.
+
+  Defaults to ``1``. Maximum of ``4000``. Only present for text inputs.
+  """
+  @type max_length :: 1..4000 | nil
+
+  @typedoc """
+  Whether the component is required to be filled, defaults to `false`.
+
+  Only present for text inputs.
+  """
+  @type required :: boolean() | nil
+
+  @typedoc """
+  The current value of the component.
+
+  When creating a new component, this will be its pre-filled value if present.
+  Only present for text inputs.
+  """
+  @type value :: String.t() | nil
```

Comment:
Let's add `:since` metadata here as well:

```suggestion
  @typedoc """
  Minimum length of the input text.

  Defaults to ``0``. Maximum of ``4000``. Only present for text inputs.
  """
  @typedoc since: "0.5.1"
  @type min_length :: 0..4000 | nil

  @typedoc """
  Maximum length of the input text.

  Defaults to ``1``. Maximum of ``4000``. Only present for text inputs.
  """
  @typedoc since: "0.5.1"
  @type max_length :: 1..4000 | nil

  @typedoc """
  Whether the component is required to be filled, defaults to `false`.

  Only present for text inputs.
  """
  @typedoc since: "0.5.1"
  @type required :: boolean() | nil

  @typedoc """
  The current value of the component.

  When creating a new component, this will be its pre-filled value if present.
  Only present for text inputs.
  """
  @typedoc since: "0.5.1"
  @type value :: String.t() | nil
```